maitake/src/wait/cell.rs

```diff
@@ -103,23 +141,25 @@ impl WaitCell {
                     "state changed unexpectedly while parking!"
                 );
 
-                if let Some(prev_waker) = prev_waker {
-                    prev_waker.wake();
-                }
-
                 if let Some(waker) = waker {
                     waker.wake();
                 }
 
-                if test_dbg!(state.is(State::CLOSED)) {
```

Comment:
I don't *think* this logic can ever happen with the given states, and returning "notified" didn't make sense to me. Let me know if I screwed some logic up.